image-loader/app/model/Uploader.scala

```diff
@@ -0,0 +1,340 @@
+package model
+
+import java.io.File
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.util.UUID
+
+import com.gu.mediaservice.lib.argo.ArgoHelpers
+import com.gu.mediaservice.lib.auth.Authentication
+import com.gu.mediaservice.lib.auth.Authentication.Principal
+import com.gu.mediaservice.lib.{StorableBrowserViewableImage, StorableOptimisedImage, StorableOriginalImage, StorableThumbImage}
+import com.gu.mediaservice.lib.aws.{S3Object, UpdateMessage}
+import com.gu.mediaservice.lib.cleanup.{MetadataCleaners, SupplierProcessors}
+import com.gu.mediaservice.lib.config.MetadataConfig
+import com.gu.mediaservice.lib.formatting._
+import com.gu.mediaservice.lib.imaging.ImageOperations
+import com.gu.mediaservice.lib.logging._
+import com.gu.mediaservice.lib.metadata.{FileMetadataHelper, ImageMetadataConverter}
+import com.gu.mediaservice.model._
+import lib.{DigestedFile, ImageLoaderConfig, Notifications}
+import lib.imaging.{FileMetadataReader, MimeTypeDetection}
+import lib.storage.ImageLoaderStore
+import model.Uploader.{fromUploadRequestShared, toImageUploadOpsCfg}
+import model.upload.{OptimiseOps, OptimiseWithPngQuant, UploadRequest}
+import org.joda.time.DateTime
+import play.api.libs.json.{JsObject, Json}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class ImageUpload(uploadRequest: UploadRequest, image: Image)
+
+case object ImageUpload {
+  val metadataCleaners = new MetadataCleaners(MetadataConfig.allPhotographersMap)
+
+  def createImage(uploadRequest: UploadRequest, source: Asset, thumbnail: Asset, png: Option[Asset],
+                  fileMetadata: FileMetadata, metadata: ImageMetadata): Image = {
+    val usageRights = NoRights
+    Image(
+      uploadRequest.imageId,
+      uploadRequest.uploadTime,
+      uploadRequest.uploadedBy,
+      Some(uploadRequest.uploadTime),
+      uploadRequest.identifiers,
+      uploadRequest.uploadInfo,
+      source,
+      Some(thumbnail),
+      png,
+      fileMetadata,
+      None,
+      metadata,
+      metadata,
+      usageRights,
+      usageRights,
+      List(),
+      List()
+    )
+  }
+}
+
+case class ImageUploadOpsCfg(
+  tempDir: File,
+  thumbWidth: Int,
+  thumbQuality: Double,
+  transcodedMimeTypes: List[MimeType],
+  originalFileBucket: String,
+  thumbBucket: String
+)
+
+case class ImageUploadOpsDependencies(
+                                       config: ImageUploadOpsCfg,
```

Comment:
minor: So much whitespace! Can we be consistent here? There's a setting in IntelliJ to control this, if only I could remember which it is...

Reply:
Not one for this PR, but it would be great to add [Scalafmt](https://scalameta.org/scalafmt/) or similar to the project.